buildspec.yml

```diff
@@ -1,7 +1,65 @@
+        go test -race -v -cover github.com/aws/amazon-ecs-cli/ecs-cli/modules/...
   build:
     commands:
-      - make integ-test
-      - ./bin/local/ecs-cli.test
\ No newline at end of file
+      - echo "Compilation context:"
+      - echo "CODEBUILD_SOURCE_VERSION=$CODEBUILD_SOURCE_VERSION"
+      # set the version number to the tag on HEAD or "development" if HEAD is
+      # not tagged
+      - VERSION=`git tag --points-at HEAD`
+      - |
+        if [ -z "$VERSION" ]; then
+          VERSION='development'
+        fi
+      - echo "VERSION=$VERSION"
+      - GIT_COMMIT_ID=`git rev-parse HEAD`
+      - echo "GIT_COMMIT_ID=$GIT_COMMIT_ID"
+      - GIT_SHORT_HASH=`git rev-parse --short=7 HEAD`
+      - echo "GIT_SHORT_HASH=$GIT_SHORT_HASH"
+      - echo "GOPATH=$GOPATH"
+
+      - GOOS=windows GOARCH=amd64 CGO_ENABLED=0 go build -installsuffix cgo -a -ldflags "-s -X github.com/aws/amazon-ecs-cli/ecs-cli/modules/version.Version=$VERSION -X github.com/aws/amazon-ecs-cli/ecs-cli/modules/version.gitShortHash=$GIT_SHORT_HASH" -o aws/amazon-ecs-cli/ecs-cli-windows-amd64-$VERSION.exe github.com/aws/amazon-ecs-cli/ecs-cli/
+      - GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -installsuffix cgo -a -ldflags "-s -X github.com/aws/amazon-ecs-cli/ecs-cli/modules/version.Version=$VERSION -X github.com/aws/amazon-ecs-cli/ecs-cli/modules/version.gitShortHash=$GIT_SHORT_HASH" -o aws/amazon-ecs-cli/ecs-cli-linux-amd64-$VERSION github.com/aws/amazon-ecs-cli/ecs-cli/
+      - GOOS=darwin GOARCH=amd64 CGO_ENABLED=0 go build -installsuffix cgo -a -ldflags "-s -X github.com/aws/amazon-ecs-cli/ecs-cli/modules/version.Version=$VERSION -X github.com/aws/amazon-ecs-cli/ecs-cli/modules/version.gitShortHash=$GIT_SHORT_HASH" -o aws/amazon-ecs-cli/ecs-cli-darwin-amd64-$VERSION github.com/aws/amazon-ecs-cli/ecs-cli/
+    finally:
+      - echo "built artifacts:"
+      - ls -lah aws/amazon-ecs-cli/
+      - ./aws/amazon-ecs-cli/ecs-cli-linux-amd64-$VERSION --version
+  post_build:
+    commands:
+      - echo "Creating latest artifacts..."
+      - cp aws/amazon-ecs-cli/ecs-cli-windows-amd64-$VERSION.exe aws/amazon-ecs-cli/ecs-cli-windows-amd64-latest.exe
+      - cp aws/amazon-ecs-cli/ecs-cli-linux-amd64-$VERSION aws/amazon-ecs-cli/ecs-cli-linux-amd64-latest
+      - cp aws/amazon-ecs-cli/ecs-cli-darwin-amd64-$VERSION aws/amazon-ecs-cli/ecs-cli-darwin-amd64-latest
+      - MANIFESTFILE="$GIT_COMMIT_ID.manifest"
+      - echo "aws/amazon-ecs-cli/ecs-cli-windows-amd64-$VERSION.exe" >> $MANIFESTFILE
+      - echo "aws/amazon-ecs-cli/ecs-cli-linux-amd64-$VERSION" >> $MANIFESTFILE
+      - echo "aws/amazon-ecs-cli/ecs-cli-darwin-amd64-$VERSION" >> $MANIFESTFILE
+      - echo "aws/amazon-ecs-cli/ecs-cli-windows-amd64-latest.exe" >> $MANIFESTFILE
+      - echo "aws/amazon-ecs-cli/ecs-cli-linux-amd64-latest" >> $MANIFESTFILE
+      - echo "aws/amazon-ecs-cli/ecs-cli-darwin-amd64-latest" >> $MANIFESTFILE
+    finally:
+      - ls -lah aws/amazon-ecs-cli/
```

Comment:
Is this just to display the artifacts? are we actually validating that we have the correct number of them like we do in our current release process?

Reply:
I suppose you are referring to the signing step in our current release process where we validate the number of signed artifacts matches the number of built artifacts? That's checked in in an internal repository. I can point you to that.